frontend/packages/ceph-storage-plugin/src/components/odf-resources/resource-details-page.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { RouteComponentProps } from 'react-router';
+import { Kebab, navFactory, LoadingBox } from '@console/internal/components/utils';
+import { OperandDetails } from '@console/operator-lifecycle-manager/src/components/operand';
+import { referenceForModel, nameForModel } from '@console/internal/module/k8s';
+import { DetailsPage } from '@console/internal/components/factory';
+import { ResourceEventStream } from '@console/internal/components/events';
+import {
+  ClusterServiceVersionModel,
+  ClusterServiceVersionKind,
+} from '@console/operator-lifecycle-manager';
+import { CustomResourceDefinitionModel } from '@console/internal/models';
+import { useK8sModel } from '@console/shared/src/hooks/useK8sModel';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { CEPH_STORAGE_NAMESPACE } from '../../constants';
+
+const csvResource = {
+  kind: referenceForModel(ClusterServiceVersionModel),
+  namespaced: true,
+  namespace: CEPH_STORAGE_NAMESPACE,
+  prop: 'csv',
+  isList: true,
+};
+
+export const GenericDetailsPage: React.FC<GenericDetailsPageProps> = (props) => {
+  const { t } = useTranslation();
+  const [model] = useK8sModel(props.match.params.resourceKind);
+
+  const crdResource = React.useMemo(
+    () => ({
+      kind: CustomResourceDefinitionModel.kind,
+      name: nameForModel(model),
+      isList: false,
+      prop: 'crd',
+    }),
+    [model],
+  );
+
+  const [csvItems] = useK8sWatchResource<ClusterServiceVersionKind[]>(csvResource);
+  const [crd] = useK8sWatchResource(crdResource);
+
+  const ocsCSV = csvItems.find((item) => item.metadata.name.includes('ocs-operator'));
```

Comment:
There is a useFetchCsv hook . It relies on spec.name of subscription to fetch exact csv.
For ocs-operator the spec.name will be "ocs-operator". You dont need to fetch all csv
It also provides loading and error states which you can use here to handle loading and error states.

Reply:
`useFetchCsv` performs 2 calls( One list call to subscriptions and then again fetching a csv). I am getting it done in one network call. I find this efficient.

Reply:
> `useFetchCsv` performs 2 calls( One list call to subscriptions and then again fetching a csv). I am getting it done in one 

yes its a generic and right way to fetch  an operator's csv, when we dont know csv name.
The `spec.name` in subscription never changes , but metadata.name is not confirmed.

> network call. I find this efficient. Subscriptions are non-namespaced and are somewhat huge payloads; IIRC I noted them 

subscriptions are not namespaced. We can modity the hook to take in ns in which csv can be found. \

> being a few MBs during early experimentation. Compared to that I would rather fetch CSV from only one namespace.

 I think it depends on how many subscriptions you have. The same goes with fetching all csvs in a namespace or any other resource like pvs or pvcs, in fact csv are bigger than subs.

Reply:
If we are sure opertaor name will always have "ocs-operator" then its fine, otherwise it would break.